service/tablet_allocator.cc

```diff
@@ -1149,6 +1150,32 @@ class load_balancer {
     }
 };
 
+future<tablet_replica_calculation_result>
+calculate_tablet_replicas_for_new_rf(const tablet_aware_replication_strategy* rep, schema_ptr s, token_metadata_ptr tm,
+    std::map<sstring, sstring> ks_options) {
+
+    auto table_id = s->id();
+    tablet_map old_tablets = tm->tablets().get_tablet_map(table_id);
+
+    if (dynamic_cast<const locator::network_topology_strategy*>(rep) == nullptr) {
```

Comment:
That should be an internal error.
We shouldn't get here otherwise.

Reply:
I'm going to remove this extra check. You're right this is validated elsewhere already + it's going to simplify the interface, as we won't need to pass `rep` any longer. I'm also going to remove this status enum altogether.